### ISSUE TITLE
Incrementally construct JSON query response

### ISSUE THREAD

**Issue by AndreasVoellmy** [graphql-java/graphql-java#1197]
Does graphql-java support  constructing the JSON for the query result incrementally, that is without actually constructing the entire object representing the result and then serializing it to JSON?

It seems like it would be possible; as the resolvers traverses through each level, it adds a field name or opens an object, then asks sub-resolvers to do the same, and then closes the object. Eventually, leaf resolves just write their values out as some primitive JSON type. Something like Jackson's streaming API (https://github.com/FasterXML/jackson-docs/wiki/JacksonStreamingApi) could be used for that.

If graphql-java does not support this, is there some way to implement that functionality using the given framework?

**Comment by bbakerman** [comment]
So nominally the spec of graphql outlines that the returned results MUST be held in memory before being released.  One reason is that of parent null ness.  If a non null child field turned out to be null then the previous parent result (according to the spec) should be made null (and that applies all the way back up the tree).

This would be impossible in a streamed version.  However one could imagine that a streaming consumer would be happy to relax this aspect.

There are some extensions in graphql that allows partial results, although technically not pure streaming.

There is a @defer directive you can use in queries

https://graphql-java.readthedocs.io/en/latest/defer.html

```
query {
   post {
       postText
       comments @defer {
           commentText
       }
       reviews @defer {
           reviewText {
       }
}
```

This will send back the postText first (complete not streamed) and then the comments down next and then the reviews.

This allows you to progressively get the most important results first and then other ones.  Again its not streaming but its trying to allow partial and early results

There is also a PR up from someone in the community that wants to do quasi streamed lists.

https://github.com/graphql-java/graphql-java/pull/1157

Its a form of sending partial results and nominally could be streamed.   But you can read me concerns on that PR.

So in summary, graphql-java does not allow stream of results, namely to be spec compliant and partly because we haven't built it like that.

It would be possible to write your own StreamingExecutionStrategy say if you really wanted to and preserve much of the query parsing and API.  But it would be its own implementation.

**Comment by AndreasVoellmy** [comment]
Great! Can you post a link to your PR?

**Comment by bbakerman** [comment]
https://github.com/graphql-java/graphql-java/pull/1204

**Comment by andimarek** [comment]
closing this issue: while this is an interesting, we don't have intermediate plans to add it.
